src/NUnitEngine/Portable/nunit.portable.agent/ResultSummary.cs

```diff
@@ -0,0 +1,342 @@
+        /// Category or Explicit attributes are not counted.
+        /// </summary>
+        public int TestCount { get; private set; }
+
+        /// <summary>
+        /// Returns the number of test cases actually run.
+        /// </summary>
+        public int RunCount
+        {
+            get { return PassCount + FailureCount + ErrorCount + InconclusiveCount; }
+        }
+
+        /// <summary>
+        /// Returns the number of test cases not run for any reason.
+        /// </summary>
+        public int NotRunCount
+        {
+            get { return IgnoreCount + ExplicitCount + InvalidCount + SkipCount; }
+        }
+
+        /// <summary>
+        /// Returns the number of failed test cases (including errors and invalid tests)
+        /// </summary>
+        public int FailedCount
+        {
+            get { return FailureCount + InvalidCount + ErrorCount; }
+        }
+
+        /// <summary>
+        /// Returns the sum of SkipCount test cases, including ignored and explicit tests
+        /// </summary>
+        public int TotalSkipCount
+        {
+            get { return SkipCount + IgnoreCount + ExplicitCount; }
+        }
+
+        /// <summary>
+        /// Gets the count of passed tests
+        /// </summary>
+        public int PassCount { get; private set; }
+
+        /// <summary>
+        /// Gets the count of failed tests, excluding errors and invalid tests
+        /// </summary>
+        public int FailureCount { get; private set; }
+
+        /// <summary>
+        /// Returns the number of test cases that had an error.
+        /// </summary>
+        public int ErrorCount { get; private set; }
+
+        /// <summary>
+        /// Gets the count of InconclusiveCount tests
+        /// </summary>
+        public int InconclusiveCount { get; private set; }
+
+        /// <summary>
+        /// Returns the number of test cases that were not runnable
+        /// due to errors in the signature of the class or method.
+        /// Such tests are also counted as Errors.
+        /// </summary>
+        public int InvalidCount { get; private set; }
+
+        /// <summary>
+        /// Gets the count of SkipCount tests, excluding ignored and explicit tests
+        /// </summary>
+        public int SkipCount { get; private set; }
+
+        /// <summary>
+        /// Gets the count of ignored tests
+        /// </summary>
+        public int IgnoreCount { get; private set; }
+
+        /// <summary>
+        /// Gets the count of tests not run because the are Explicit
+        /// </summary>
+        public int ExplicitCount { get; private set; }
+
+        /// <summary>
+        /// Gets the count of invalid assemblies
+        /// </summary>
+        public int InvalidAssemblies { get; private set; }
+
+        /// <summary>
+        /// An Unexpected error occurred
+        /// </summary>
+        public bool UnexpectedError { get; private set; }
+
```

Comment:
Should there be something to handle this written to the xml? Would an unexpected error currently be lost and return a failure for no clear reason? I'm entirely clear on what's upstream from this.